google/cloud/storage/internal/curl_resumable_streambuf.cc

```diff
@@ -92,48 +92,65 @@ std::streamsize CurlResumableStreambuf::xsputn(char const* s,
 
 StatusOr<HttpResponse> CurlResumableStreambuf::DoClose() {
   GCP_LOG(INFO) << __func__ << "()";
-  return Flush(true);
+  return FlushFinal();
 }
 
-StatusOr<HttpResponse> CurlResumableStreambuf::Flush(bool final_chunk) {
+StatusOr<HttpResponse> CurlResumableStreambuf::FlushFinal() {
+  // Shorten the buffer to the actual used size.
+  auto actual_size = static_cast<std::size_t>(pptr() - pbase());
+  std::size_t upload_size = upload_session_->next_expected_byte() + actual_size;
+  current_ios_buffer_.resize(actual_size);
+  hash_validator_->Update(current_ios_buffer_);
+
+  StatusOr<ResumableUploadResponse> result;
+  result = upload_session_->UploadFinalChunk(current_ios_buffer_, upload_size);
+  if (!result.ok()) {
```

Comment:
if (!result) ... ?

Reply:
Done.

Reply:
I don't see that one (but I do see a similar update below).

Reply:
Doh, I think I did it right this time.